gadget/install/partition.go

```diff
@@ -70,7 +70,7 @@ type CreateOptions struct {
 // OnDiskStructure, as it is meant to be used externally (i.e. by
 // muinstaller).
 func CreateMissingPartitions(dv *gadget.OnDiskVolume, gv *gadget.Volume, opts *CreateOptions) ([]*gadget.OnDiskAndGadgetStructurePair, error) {
-	dgpairs, err := createMissingPartitions(dv, gv, opts)
+	dgpairs, err := createMissingPartitions(dv, gv, opts, map[int]StructOffsetSize{})
```

Comment:
as we don't write to it this can be nil afaict

Reply:
Good catch, I had forgotten that maps can be nil if we are only reading.